src/Selenium2Driver.php

```diff
@@ -1120,4 +1132,52 @@ private function trigger($xpath, $event, $options = '{}')
         $script = 'Syn.trigger("' . $event . '", ' . $options . ', {{ELEMENT}})';
         $this->withSyn()->executeJsOnXpath($xpath, $script);
     }
+
+    /**
+     * Uploads a file to the Selenium instance.
+     *
+     * Note that uploading files is not part of the official WebDriver
+     * specification, but it is supported by Selenium.
+     * @see https://github.com/SeleniumHQ/selenium/blob/master/py/selenium/webdriver/remote/webelement.py#L533
+     *
+     * @param string $path     The path to the file to upload.
+     *
+     * @return string          The remote path.
+     *
+     * @throws DriverException When the file is not found.
+     * @throws InvalidRequest  When the driver does not support file uploads.
+     */
+    public function uploadFile($path) {
+        if (!is_file($path)) {
+          throw new DriverException('Could not upload file, the file: ' . $path . '. was not found.');
+        }
+
+        // Selenium only accepts uploads that are compressed as a Zip archive.
+        $temp_filename = tempnam('', 'WebDriverZip');
+
+        $archive = new \ZipArchive();
```

Comment:
Yes that is what is going to happen now. If `ZipArchive` is not there a `DriverException` is thrown, and this is caught by `attachFile()` which will then continue with the original (local) path. It will then fail exactly as it does now later on when the `sendKeys` command is executed.